src/metadata.rs

```diff
@@ -319,6 +319,20 @@ impl ModelMetadata {
     pub fn class_name(&self, class_id: usize) -> Option<&str> {
         self.names.get(&class_id).map(String::as_str)
     }
+
+    /// Extract the model name from the description.
+    ///
+    /// E.g. "Ultralytics `YOLO11n` model..." -> "`YOLO11n`"
+    /// Returns "YOLO" if extraction fails.
+    #[must_use]
+    pub fn model_name(&self) -> String {
```

Comment:
❗ **CRITICAL**: Current extraction will fail for the documented example (e.g. "Ultralytics `YOLO11n` model...") because the token starts with a backtick, so `to_lowercase().starts_with("yolo")` won’t match. It also allocates a new `String` for every word via `to_lowercase()`. Consider trimming surrounding punctuation/backticks and doing an ASCII case-insensitive prefix check without allocation.

**Suggested change:**
```suggestion
    pub fn model_name(&self) -> String {
        self.description
            .split_whitespace()
            .map(|w| w.trim_matches(|c: char| !c.is_ascii_alphanumeric()))
            .find(|w| w.get(..4).map_or(false, |p| p.eq_ignore_ascii_case("yolo")))
            .unwrap_or("YOLO")
            .to_string()
    }
```